News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/database/DatabaseConnectionOrm.java

```diff
@@ -327,13 +327,20 @@ public boolean doesRssItemAlreadyExsists (long feedId) {
         return feeds.size() > 0;
     }
 
-
     public void removeFeedById(long feedId) {
         daoSession.getFeedDao().deleteByKey(feedId);
-    }
-
 
+        List<RssItem> list = daoSession.getRssItemDao().queryBuilder().where(RssItemDao.Properties.FeedId.eq(feedId)).list();
+        for (RssItem rssItem : list) {
+            daoSession.getRssItemDao().delete(rssItem);
```

Comment:
I think we can wrap the delete operation in a single transaction. This should give us a huge performance improvement if there are a lot RssItems for a feed.